### PR TITLE
testdrive: Benchmarking mode

### PR DESCRIPTION
**Pull request by andrewrodriguez-m**
#22033 for motivation: testdrive is sometimes used for benchmarking and sometimes for just testing. This flag distinguishes the two cases

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by jkosh44 — APPROVED** [review]

File: src/testdrive/src/action/sql.rs [hunk 1]
```diff
@@ -122,90 +122,92 @@ pub async fn run_sql(mut cmd: SqlCommand, state: &mut State) -> Result<ControlFl
...
-        | Statement::RevokePrivileges { .. }
-        | Statement::RevokeRole { .. } => {
-            let response = Retry::default()
-                .max_duration(Duration::from_secs(3))
-                .clamp_backoff(Duration::from_millis(500))
-                .retry_async(|_| async {
-                    reqwest::get(&format!(
-                        "http://{}/api/coordinator/check",
+    if !state.benchmarking {
+        match stmt {
```
Comment (nrainer-materialize) [inline src/testdrive/src/action/sql.rs:126]: This is a rather long method. What about extracting the body of this if-condition to a method? What do you think?

Comment (andrewrodriguez-m) [reply src/testdrive/src/action/sql.rs]: Sure, done

**Review by nrainer-materialize — APPROVED** [review]

**Review by def- — APPROVED** [review]
I triggered feature benchmark on this change, let's see if this fixes it: https://buildkite.com/materialize/nightlies/builds/4463

**Review by def- — CHANGES_REQUESTED** [review]
Could we rename this to --no-consistency-check and also add it in Cloud Canary tests? I think this would fix https://github.com/MaterializeInc/materialize/issues/22056

**Review by def- — APPROVED** [review]
Thanks Joe

**Pull request merged**